core/src/scene/styleParam.cpp

```diff
@@ -307,105 +298,115 @@ StyleParam::Value StyleParam::parseString(StyleParamKey key, const std::string&
     case StyleParamKey::priority:
     case StyleParamKey::text_max_lines:
     case StyleParamKey::text_priority: {
-        int num;
-        if (parseInt(_value, num) > 0) {
-            if (num >= 0) {
-                return static_cast<uint32_t>(num);
-            }
+        int result = -1;
+        if (YamlUtil::getInt(node, result)) {
+            return static_cast<uint32_t>(result);
         }
-        LOGW("Invalid '%s' value '%s'", keyName(key).c_str(), _value.c_str());
+        LOGW("Invalid '%s' value '%s'", keyName(key).c_str(), Dump(node).c_str());
         break;
     }
     case StyleParamKey::placement_spacing: {
-        ValueUnitPair placementSpacing;
-        placementSpacing.unit = Unit::pixel;
-
-        int pos = parseValueUnitPair(_value, 0, placementSpacing);
-        if (pos < 0) {
-            LOGW("Invalid placement spacing value '%s'", _value.c_str());
-            placementSpacing.value =  80.0f;
-            placementSpacing.unit = Unit::pixel;
-        } else {
-            if (placementSpacing.unit != Unit::pixel) {
-                LOGW("Invalid unit provided for placement spacing");
+        ValueUnitPair result;
+        if (node.IsScalar() && parseValueUnitPair(node.Scalar(), result)) {
+            if (result.unit == Unit::none) {
+                result.unit = Unit::pixel;
             }
         }
-
-        return Width(placementSpacing);
+        if (result.unit != Unit::pixel) {
+            LOGW("Invalid placement spacing value '%s'", Dump(node).c_str());
+            result.value = 80.f;
```

Comment:
I think I initially added the `80` default value here. Should have added a const or a define for it. Can you take care of it here?